DEV-4622: Update data-validator configurations and improve Holesky setup

eoracle-ak-47 opened the pull request:
Added new environment variables to enhance logging, custom ports, and configuration flexibility for the data-validator. Updated `.example_env_holesky` with additional setup details, such as addresses and endpoints, while refining Holesky-specific instructions in `README.md`.

eoracle-ak-47 committed:
Update data-validator configurations and improve Holesky setup

Added new environment variables to enhance logging, custom ports, and configuration flexibility for the data-validator. Updated `.example_env_holesky` with additional setup details, such as addresses and endpoints, while refining Holesky-specific instructions in `README.md`.

uri-lightblocks commented on README.md:
```diff
@@ -28,8 +28,9 @@ cp data-validator/.example_env data-validator/.env
-If you plan to operate eoracle on Holesky, then copy `.example_env_holeksy` instead of `.example_env`
+[**NOTE**]
+If you plan to operate eoracle on Holesky, then copy `.example_env_holeksy` instead of `.example_env` and current 
+process of registration can be ommited since you'll need contact support to register for now.  
```
A new operator still requires to register
And then follow up the instructions to execute.
An existing operator just need to upgrade.
Maybe add a section to upgrading from v1 to v2. At the moment it relevant only for the Holesky environment

uri-lightblocks commented on data-validator/.example_env:
This file should not be changed, as it for v1

eoracle-ak-47 replied:
true but this envs used in docker-compose which is for both env files so this changes are to keep consistency

uri-lightblocks commented on data-validator/.example_env_holesky:
The first lines are required to support registration process

eoracle-ak-47 replied:
true, but we are changing config contract address so commands below woldn work for now
- print-status
- declare-alias
- reset-configuration

uri-lightblocks replied:
they will work since we have the v1 contracts

eoracle-ak-47 committed:
Add Eoracle configuration variables to example env file

This update includes essential configuration variables for Eoracle integration, such as RPC endpoint, registry coordinator, stake registry, and aggregator address. These additions ensure smoother setup for users working with the Holesky testnet.

eoracle-ak-47 committed:
Fix Holesky-related instructions in README.md

Clarified setup steps for operating on Holesky by removing outdated references and adding notes about unsupported commands. Updated guidance highlights the need to contact support for certain actions.

uri-lightblocks commented on data-validator/.example_env_holesky:
```diff
@@ -1,34 +1,99 @@
-EO_OPERATOR_CLI_IMAGE=eoracle/opr_cli:v0.0.12
-EO_DATA_VALIDATOR_IMAGE=eoracle/data-validator:latest
 EO_CHAIN_RPC_ENDPOINT=https://rpc.testnet.eoracle.network
```
move this to be near `EO_CHAIN_WS_ENDPOINT`

uri-lightblocks replied:
bumping

uri-lightblocks commented on data-validator/.example_env_holesky:
```diff
@@ -1,34 +1,99 @@
+# eOracle chain config contract address
+EO_CONFIG_ADDRESS=0xa72c694e619dd339ab33415e07f0afc793555053
+
+# eOracle operator main address
+EO_MAIN_ADDRESS=enterYourMainAddressHere
+# eOracle OVS key mapping as a list of wallet addresses used to sign transactions for the OVS,
+# paired with their corresponding OVS addresses.
+# Each pair is separated by a colon and multiple pairs are separated by commas.
+# (example: 0xSignerWalletAddress1:0xOvsAddress1,0xSignerWalletAddress2:0xOvsAddress2)
+EO_OVS_KEY_MAPPING=enterYourAliasAddressHere:0x530fCc836C42a803146d48be8Af295bA9756E266
```
We need to ensure operators will know they must enter their details here.
I suggest moving this section near the passphrase

uri-lightblocks commented on data-validator/.example_env_holesky:
```diff
@@ -1,34 +1,99 @@
+
+# eOracle operator main address
+EO_MAIN_ADDRESS=enterYourMainAddressHere
+# eOracle OVS key mapping as a list of wallet addresses used to sign transactions for the OVS,
+# paired with their corresponding OVS addresses.
+# Each pair is separated by a colon and multiple pairs are separated by commas.
+# (example: 0xSignerWalletAddress1:0xOvsAddress1,0xSignerWalletAddress2:0xOvsAddress2)
```
Keep it simple for now, and remove the multiple pairs example

uri-lightblocks commented on data-validator/.example_env_holesky:
```diff
@@ -1,34 +1,99 @@
+# Endpoint of the coinbase source (default: "wss://ws-feed.exchange.coinbase.com")
+# ENDPOINT_COINBASE=
+# Endpoint of the coingecko source (default: "https://pro-api.coingecko.com/api/v3")
+# ENDPOINT_COINGECKO=
+# Auth key for the coingecko source
+# ENDPOINT_COINGECKO_AUTH=
+# Endpoint of the coinmarketcap source (default: "https://pro-api.coinmarketcap.com/v2/cryptocurrency/quotes/latest")
```
replace the order as you have the auth before the endpoint in other places

eoracle-ak-47 committed:
Update documentation and configuration for Data Validator V2

Updated the README to include details on logging and setup for Data Validator V2. Enhanced `.example_env_holesky` with new required configurations and improved structure to support the updated system requirements.

eoracle-ak-47 committed:
Update README to remove outdated Holesky support notes

Removed notes about Holesky incompatibility for `declare-alias` and `print-status` commands as they are no longer relevant. This streamlines the documentation and avoids potential confusion for users.

eoracle-ak-47 committed:
Update data-validator image to use v2-latest tag

Switched the EO_DATA_VALIDATOR_IMAGE from a specific version to the v2-latest tag for more flexibility in deploying the latest updates. This ensures the image always points to the most recent stable release within the v2 series.

uri-lightblocks commented on data-validator/.example_env:
```diff
@@ -9,9 +9,11 @@ EO_CONFIG_ADDRESS=0x05a6f762f64Ac2ccE0588677317a0Ed8af9d0c16
 ETH_RPC_ENDPOINT=https://rpc.flashbots.net
 EO_PASSPHRASE=enterYourPassphraseHere
 EO_KEYSTORE_PATH=.keystore
+EO_DATA_PORT=10005
```
remove this line

uri-lightblocks commented on data-validator/.example_env:
```diff
@@ -9,9 +9,11 @@ EO_CONFIG_ADDRESS=0x05a6f762f64Ac2ccE0588677317a0Ed8af9d0c16
 EO_PROMETHEUS_PORT=10004
 EO_HEALTH_ENDPOINTS_PORT=10003
 EO_LOGGING_LEVEL=info
+EO_LOG_OUT_PATH=./logs
```
remove this line

uri-lightblocks commented on data-validator/.example_env_holesky:
```diff
@@ -1,34 +1,113 @@
+EO_CONFIG_ADDRESS=0xa72c694e619dd339ab33415e07f0afc793555053
+
+# Endpoint of the Ethereum feed RPC (default: "wss://eth.llamarpc.com")
+ENDPOINT_ETHEREUM_FEED_RPC=wss://ethereum-holesky-rpc.publicnode.com
```
This should be mainnet

eoracle-ak-47 committed:
Remove deprecated and duplicate environment variables

Cleaned up unused variables and relocated EO_CHAIN_RPC_ENDPOINT for consistency in the `.example_env` and `.example_env_holesky` files. Updated Ethereum feed RPC endpoint to the correct public node URL.
